### ISSUE TITLE
STAS-F01: Missing ‘Skip’ Navigation

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/MyopicVicar#917]
The purpose of skip navigation is to provide a mechanism to bypass blocks of material that are
repeated on multiple Web pages by skipping directly to the main content of the Web page. One of
the first interactive items on a web page should be a link to the beginning of the main content.
Activating the link sets focus beyond the repeated content to the main content of the page.
If there is no „Skip‟ Navigation present, screen reader users would have to listen to content on
pages visited on the website, and keyboard only users would have to tab through all the links until
they arrive at the main content of the page. Ideally, the „skip‟ to content link should take the user to
just above the header of the main content at the top left hand side.
There is currently no evidence of a skip to content link present on the site.

**Comment by AlOneill** [comment]
We have some decisions to make: whatever the actual approach, it should be consistent across pages.

1. Skiplink needs to be at top of page visually and by source order, so that it has focus with the first press of tab. However, it does not need to be visible unless it has focus. There are pros and cons of each.

2. Depending on the page, the (very) top of the main content may not always be the best destination. For instance, the page menu on (researcher) Help could be skipped by placing that menu first in the page 'body' in the source order, with the link destination being after it. Alternatively, a page could have more than one skiplink, allowing for skipping over the page menu itself. (Will continue to investigate practice on accessible sites.)

3. We also need to consider exactly where to direct any 'back to top' / 'back to ...' type links.

This issue is related to #929 (Illogical tab order), #930 (Links not visible in focus) and #932 (Visual cue on tab focus)

**Comment by nurifreembd** [comment]
@Denise Colbert ***@***.***>  Would you like me
to arrange a workshop? If yes, who needs to be involved in discussing,
agreeing or disagreeing the feature please? If agreed, prioritise it for
development in future road map and or increment or sprint.

If agreed we would need a design to show how it would look and where it
would be housed on the page.

Many Thanks,

**Comment by nurifreembd** [comment]
@Denise Colbert ***@***.***> happy to discuss
with the Accessibility team as all sites are impacted in our Friday
catch-up or in a separate call. It would be good to have you and a
developer in the call. a developer to get techinical input. Do let me know
what your preference is to move the skip feature forward to a resolution.

**Comment by nurifreembd** [comment]
Attached Skip navigation User Story. 

[Skip Navigation Solution V.01.docx](https://github.com/FreeUKGen/MyopicVicar/files/14538717/Skip.Navigation.Solution.V.01.docx)

**Comment by DeniseColbert** [comment]
Ready for testing in Test3 @nurifreembd would you be able to test this?

**Comment by nurifreembd** [comment]
@denisecolbert, Yes. @Vino-S can you provide access to Test3 please?

**Comment by Vino-S** [comment]
Thanks @AlOneill  for testing.
@nurifreembd , updated test3: Renamed Skip to **Skip to content**. Below is the screenshot:
![image](https://github.com/FreeUKGen/MyopicVicar/assets/26073943/17a75d1c-0440-4af1-a01c-f0c973abc481)


_This implementation gives no immediate indication that anything will be skipped. In fact it looks like the ad has already been skipped because there is no reason for a user to understand the strange source order at the top of our pages!_
Currently, when user tab to the **Skip to content** link skips the header advert and its form fields and the project Logo.
Should I add the Skip link at the beginning of the Navigation bar as in BBC and thus skipping the Navigation bar links.

**Comment by nurifreembd** [comment]
Thank you Vino. Can we agree the link to be renamed to 'Skip Advert'
instead please? Denise would you agree?

I'll try and test as I am having trouble with my browser.

**Comment by nurifreembd** [comment]
Vino,

Can you apply skip navigation to Freereg and Freecen too please?

**Comment by AlOneill** [comment]
I do not think that "Skip advert" is a good idea because links should state the _destination_, so "skip to something" rather than "skip past something".

Also why deploy something that is not yet correct to a second website?? Let's walk before we try running.

I have no more energy today.

**Comment by nurifreembd** [comment]
Alison,

We had a discussion around this and the functional requirement was agreed
with Denise and Vino. Attached is the user story. If you disagree with this
agreement please discuss with Denise as Vino is coding what was agreed. We
should not deviate from the agreed requirement unless agreed with all.

Vino, Denise, please confirm if we should go with the agreed user story as
discussed or open it up for further discussion to avoid wasting Vino's dev
time?

Vino, I have tried to test across all 3 sites on test3 but still not
working for me. Can you help please?

**Comment by DeniseColbert** [comment]
Our decision was made without Alison's input because she isn't able to come
to scrums due to illness, Nuri. @Vinodhini Subbu
***@***.***> suspend work on this until we can
review Alison's suggestion in the scrum, please.

Alison's point about not deploying to other sites until the fix is working
is valid; we only transfer once testing is complete and the fix is live on
the site it was completed on.

If you can tell me exactly what is not working for you on Test3 Nuri, Vino
and/or I should be able to help. Please note, you will only see the skip
link in the nav bar if you tab to it via your keyboard.


Best wishes,

Denise


Denise Colbert
*Chief Operating Officer, *Free UK Genealogy CIO
Registered in England and Wales, number 1167484
<http://www.facebook.com/freeukgen> <https://twitter.com/FreeUKGen>
<https://www.instagram.com/freeukgenealogy>
<https://www.pinterest.co.uk/FreeUKGenealogy>
<https://www.linkedin.com/company/10057479>
<https://www.youtube.com/channel/UCP839R-ftWroUjqv9f02aTg>

**Comment by Vino-S** [comment]
Noted

**Comment by nurifreembd** [comment]
Sure agreed about not deploying to other sites just yet. Will update the
user story following Alison's input.

**Comment by nurifreembd** [comment]
Alison

re Let's walk before we try running.

I did not ask for it to be deployed to a production environment.
a code on one side could work fine but could break something else on
another site. I was asking to test it within a test environment.

**Comment by DeniseColbert** [comment]
@Vino-S to look at language for skipping to [destination]

**Comment by nurifreembd** [comment]
Thanks Denise.

We may have discussed the option of having an alt text on the advert to
skip or stay Ad, but cannot recall why we dismissed the idea.

story we could reconsider would be, below is a draft and for discussion
only:
User has a screen reader
user tabs through menu options
on tabbing through Help
the next focus is placed on the advert banner
user is prompted with an alt text 'skip Ad' or 'Skip Ad in 3 seconds' and
jump to the next focus in the order

If user  is using a keyboard and does not have a screen reader to navigate,
the existing functionality stays the same and the alt text prompt to skip
banner is hidden from user.

The above approach could be much neater if reconsidered and would not
distract the menu style.

I don't see anything wrong with using the word 'Skip Ad'  or 'skip ad in 3
secs' as the user is and will be skipping it or navigating it.

Many thanks,

**Comment by DeniseColbert** [comment]
@Vino-S to place code on test 3 for Nuri to test.

**Comment by DeniseColbert** [comment]
@nurifreembd can you check this works as expected please?

**Comment by AlOneill** [comment]
@nurifreembd @DeniseColbert @Vino-S  As I have remarked previously, **a screen reader user has no need to use a skip-link** — they have other, more direct ways of navigating a page. Skip-links are for sighted keyboard-only users. They expect to find a skip-link with their very first press of tab. That skip-link will state the destination, typically the beginning of the main content, the `main` element.

As far as I can see, test 3 behaves as I would expect it to: on a first visit to the **search page**, the permission for cookies is dealt with initially and on the next press of tab the skip-link is revealed. Selecting that link (using the Enter key) takes me to the main content where I can tab through the form. Perfect.

On the **About page**, however, I cannot see what has focus between the final "On this page" menu item and the first link under "Membership". This is not good and needs fixing.
